### PR TITLE
ENH: setup dependencies.json

### PR DESCRIPTION
**Pull request by anish-mudaraddi**
having a separate dependencies file makes it easier for us to update versions and potentially use dependabot or something similar for this repo

### CODE REVIEW

File: dependencies.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "capo-addon-version": "0.5.9",
+    "capo-provider-version": "v0.10.4",
```
Comment (meoflynn) [inline dependencies.json:3]: Could we update the capo provider version to `v0.11.0` ?

**Review by meoflynn — CHANGES_REQUESTED** [review]

File: dependencies.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+    "capo-addon-version": "0.5.9",
```
Comment (meoflynn) [inline dependencies.json:2]: Can we also update the capo-addon to 0.7.0?

File: dependencies.json (hunk 1 above)
Comment (anish-mudaraddi) [reply dependencies.json:3]: cluster api provider openstack - CAPO is v0.10.5 - https://github.com/kubernetes-sigs/cluster-api-provider-openstack/releases/tag/v0.10.5 - i'll update that and the addon-version which should be 0.7.0

if you're referring to azimuth-cloud/capi-helm-charts chart version - users should automatically use the most up-to-date chart when they follow our docs

File: dependencies.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "capo-addon-version": "0.7.0",
```
Comment (meoflynn) [inline dependencies.json:2]: could we change this variable name slightly? e.g `aziumuth-capo-addon`?

File: dependencies.json (hunk 1 above)
Comment (meoflynn) [reply dependencies.json]: Ah I got the variables mixed up with what they refer to 😅

**Review by gmatthews20 — APPROVED** [review]

**Comment by anish-mudaraddi** [comment]
I edited the latest commit - you didn't change the variable name in the script to match the change in dependency name - needs re-review @meoflynn

**Comment by meoflynn** [comment]
Good catch, I didn't realize I forgot to update the variable name 😅

**Review by meoflynn — APPROVED** [review]

**Review by RJaliliSTFC — APPROVED** [review]

**Pull request merged**